kover-gradle-plugin/src/main/kotlin/kotlinx/kover/gradle/plugin/dsl/KoverVariantConfig.kt

```diff
@@ -0,0 +1,241 @@
+     *
+     * Example:
+     * ```
+     *  sources {
+     *     // exclude classes compiled by Java compiler from all reports
+     *     excludeJava = true
+     *
+     *     // exclude source classes of specified source sets from all reports
+     *     excludedSourceSets.addAll(excludedSourceSet)
+     * ```
+     */
+    public fun sources(block: Action<KoverVariantClasses>)
+
+    /**
+     * Instrumentation settings for the current Gradle project.
+     *
+     * Instrumentation is the modification of classes when they are loaded into the JVM, which helps to determine which code was called and which was not.
+     * Instrumentation changes the bytecode of the class, so it may disable some JVM optimizations, slow down performance and concurrency tests, and may also be incompatible with other instrumentation libraries.
+     *
+     * For this reason, it may be necessary to fine-tune the instrumentation, for example, disabling instrumentation for problematic classes. Note that such classes would be marked as uncovered because of that.
+     *
+     * Example:
+     * ```
+     *  instrumentation {
+     *      // disable instrumentation of all classes in test tasks that are in the current project
+     *      excludeAll = true
+     *
+     *      // disable instrumentation of specified classes in test tasks that are in the current project
+     *      excludedClasses.addAll("foo.bar.*Biz", "*\$Generated")
+     *  }
+     * ```
+     */
+    public fun instrumentation(block: Action<KoverVariantInstrumentation>)
+
+    /**
+     * Set up tests, the run of which is used to measure coverage.
+     *
+     * To measure coverage, Kover runs Gradle test tasks, instrumentation takes place before they are performed, and code coverage is measured during execution.
+     *
+     * By default, Kover use all [org.gradle.api.tasks.testing.Test] to measure coverage.
+     *
+     * Example:
+     * ```
+     *  testTasks {
+     *      // The coverage of the test1 and test2 tasks will no longer be taken into account in the reports
+     *      //  as well as these tasks will not be called when generating the report
+     *      excluded.addAll("test1", "test2")
+     *  }
+     * ```
+     */
+    public fun testTasks(block: Action<KoverVariantTestTasks>)
+}
+
+/**
+ * Limit the classes that will be included in the reports.
+ * These settings do not affect the instrumentation of classes.
+ *
+ * The settings specified here affect all reports in any projects that use the current project depending on.
+ * However, these settings should be used to regulate classes specific only to the project in which this setting is specified.
+ *
+ * Example:
+ * ```
+ *  sources {
+ *     // exclude classes compiled by Java compiler from all reports
+ *     excludeJava = true
+ *
+ *     // exclude source classes of specified source sets from all reports
+ *     excludedSourceSets.addAll(excludedSourceSet)
+ * ```
+ */
+public interface KoverVariantClasses {
```

Comment:
Probably name should be `KoverVariantSources`?

Reply:
Indeed